### PR TITLE
CI: Add support for GitHub actions

### PR DESCRIPTION
**Pull request by thrasher-**
# PR Description

Adds GitHub actions support. Note: macOS/windows runners do not support postgres services so this has only been implemented for Linux runners. Builds are currently failing due to a breaking API change in our default API key free currency converter. Will create a PR next week to address this.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How has this been tested

- [x] CI output 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Travis with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        CGO_ENABLED: 1
+
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v1
+
+  frontend:
+    name: GoCryptoTrader frontend
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
```
Comment (gloriousCode) [inline .github/workflows/tests.yml:123]: Can this be added to this run?
```
   - name: Cancel previous workflow runs
      uses: styfle/cancel-workflow-action@0.8.0
      with:
        access_token: ${{ github.token }}
```
Or does it not work for front-end? Cancelling everything on a new commit is great, so I'd want it on everything

**Review by shazbert — APPROVED** [review]
Changes ACK! This is a much-needed addition. 🦅

File: .github/workflows/tests.yml (hunk 1 above)
Comment (thrasher-) [reply .github/workflows/tests.yml:128]: Yep it does work, it's a quick test compared to others that I didn't think it was necessary. However, I've added support for it anyway

**Review by gloriousCode — APPROVED** [review]
Thank you for making that change! ACK

**Review by shazbert — APPROVED** [review]
changes ACK!

**Review by MadCozBadd — APPROVED** [review]
ACK!!!!!!

**Review by gloriousCode — APPROVED** [review]

**Pull request merged**